File: swift/Sources/allmark/RenderConsole/ConsoleCodeSpanRenderer.swift
import Foundation

@MainActor
let consoleCodeSpanRenderer = Renderer(
	name: "code_span",
	render: renderConsoleCodeSpan
)

@MainActor
func renderConsoleCodeSpan(_ node: MarkdownNode, _ state: inout RendererState, _ first: Bool?, _ last: Bool?, _ decode: Bool?) {
	let styles = getConsoleStyles()
	let style = styles["code"] ?? ""
	state.output += style
	state.output += "`"
	renderChildren(node: node, state: &state)
	state.output += "`"
	state.output += ansiReset
}

File: dotnet/Allmark/RenderConsole/ConsoleCodeSpanRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ConsoleCodeSpanRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "code_span",
            Render = (node, state, decode) => Render(node, state),
        };
    }

    public static void Render(MarkdownNode node, RendererState state)
    {
        var style = Ansi.Green;
        state.Output.Append(style);
        RenderChildren.Execute(node, state);
        state.Output.Append(Ansi.Reset);
    }
}
